### ISSUE TITLE
Restructure Workspace

### ISSUE THREAD

**Issue by ChristopherPHolder** [ChristopherPHolder/app-speed#100]
## TL;DR;

Restructure workspace to improve architecture and readability.

## Description 

Currently the workspace is very badly structures, and has no documentation about the way things should be structures.

This MR will improve the general structure to reflect best practices. It will also add proper documentation into the readme and lint rules to prevent this issue from coming back.

References:
- [NX Docs](https://nx.dev/concepts/decisions/overview)
- [Greenforce Frontend Repo](https://github.com/Energinet-DataHub/greenforce-frontend) This is an example provided by Lars as a good structure.
- [Code-Pushup CLI Repo](https://github.com/code-pushup/cli)

**Comment by ChristopherPHolder** [comment]
Generally speaking there are top level approaches to folder structures: 
- `apps/libs`
- `packages/tools`

There is no right and wrong answer here but I agree with Edouard Bozon. 

Using `packages/tools` will be used to provide better code colocation.

This is already partially implemented to the repo already. But is not properly implmented as currently many things are provided in the top level `packages`. This make it very hard to read and not easy to understand what is and project of its own and what actually belongs to another project. 

For this reason we will reserve the top level to applications or packages and shared
```
packages
├──  conductor
├──  portal
├──  runner
└──  shared
```
